functions/Test-DbaCompression.ps1

```diff
@@ -0,0 +1,371 @@
+	
+	.PARAMETER SqlInstance
+		SqlInstance name or SMO object representing the SQL Server to connect to. This can be a collection and recieve pipeline input
+	
+	.PARAMETER SqlCredential
+		PSCredential object to connect under. If not specified, currend Windows login will be used.
+	
+	.PARAMETER Database
+		The database(s) to process - this list is autopopulated from the server. If unspecified, all databases will be processed.
+	
+	.PARAMETER ExcludeDatabase
+		The database(s) to exclude - this list is autopopulated from the server
+	
+	.PARAMETER IncludeSystemDBs
+		Switch parameter that when used will display system database information
+	
+	.PARAMETER Silent
+		Replaces user friendly yellow warnings with bloody red exceptions of doom!
+		Use this if you want the function to throw terminating errors you want to catch.
+	
+	.EXAMPLE
+		
+		
+		Returns all user database files and free space information for the local host
+	
+	.EXAMPLE
+		Test-DbaCompression -SqlInstance ServerA -Database DBName | Out-GridView
+		Returns results of all potential compression options for a single database 
+        with the recommendation of either Page or Row into and nicely formated GridView
+	
+	.EXAMPLE
+		Test-DbaCompression -SqlInstance ServerA 
+		Returns results of all potential compression options for all databases
+        with the recommendation of either Page or Row
+	
+    .EXAMPLE
+        $servers = 'Server1','Server2'
+
+        foreach ($svr in $servers)
+        {
+        Test-DbaCompression -SqlInstance $svr | Export-Csv -Path C:\temp\CompressionAnalysisPAC.csv -Append
+        }
+	    This produces a full analysis of all your servers listed and is pushed to a csv for you to 
+        analyize.
+
+	.NOTES
+		
+		Website: https://dbatools.io
+		Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+		License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+	
+	.LINK
+		https://dbatools.io/Test-DbaCompression
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]
+		$SqlInstance,
+		
+		[System.Management.Automation.PSCredential]
+		$SqlCredential,
+		
+		[Alias("Databases")]
+		[object[]]
+		$Database,
+		
+		[object[]]
+		$ExcludeDatabase,
+		
+		[switch]
+		$IncludeSystemDBs,
+		
+		[switch]
+		$Silent
+	)
+	
+	begin {
+		Write-Message -Level System -Message "Bound parameters: $($PSBoundParameters.Keys -join ", ")"
+		$sql = "SET NOCOUNT ON;
+
+CREATE TABLE ##tmpCompression ([Schema] sysname
+	,[Table_Name] sysname
+	,[Index_Name] sysname NULL
+	,[Partition] int
+	,[Index_ID] int
+	,[Index_Type] VARCHAR(12)
+	,[Percent_Scan] smallint
+	,[Percent_Update] smallint
+	,[ROW_estimate_Pct_of_orig] bigint
+	,[PAGE_estimate_Pct_of_orig] bigint
+	,[Compression_Type_Recommendation] VARCHAR(7)
+	,size_cur bigint
+	,size_req bigint
+);
+
+CREATE TABLE ##tmpEstimateRow (
```

Comment:
I got lazy - @js0505 - please sync development and submit a patch for the updated command. I changed it to ##testdbacompression but could still use 2008+ compatible check for exists/drop